bothan-core/src/manager/crypto_asset_info/price.rs

```diff
@@ -0,0 +1,57 @@
+use std::collections::{HashMap, HashSet};
+use std::sync::Arc;
+
+use tracing::debug;
+
+use crate::manager::crypto_asset_info::error::GetPriceError;
+use crate::manager::crypto_asset_info::price::tasks::{create_reduced_registry, execute_task_set};
+use crate::manager::crypto_asset_info::price::utils::get_price_state;
+use crate::manager::crypto_asset_info::types::PriceState;
+use crate::registry::Registry;
+use crate::tasks::TaskSet;
+use crate::worker::AssetWorker;
+
+mod tasks;
+mod utils;
+
+const PRECISION: u32 = 9;
+
+pub async fn get_prices<'a>(
+    ids: &[String],
+    registry: &Registry,
+    workers: &HashMap<String, Arc<dyn AssetWorker + 'a>>,
+    stale_threshold: i64,
+) -> Result<Vec<PriceState>, GetPriceError> {
+    let current_time = chrono::Utc::now().timestamp();
+
+    // Split the signals into those that exist and those that do not.
+    // Signal that are not available will return UNSUPPORTED
+    debug!("Processing {} signals", ids.len());
+    let (supported, unsupported): (Vec<String>, Vec<String>) = ids
+        .iter()
+        .cloned()
+        .partition(|id| registry.contains_key(id));
+
+    debug!("Supported signals: {:?}", supported);
+    debug!("Unsupported signals: {:?}", unsupported);
+
+    let unsupported_ids = unsupported.into_iter().collect::<HashSet<String>>();
+    let reduced_registry = create_reduced_registry(supported.clone(), registry)
```

Comment:
I think creating reduced registry and trying to get prices of multiple ids at once complicates the code without enough justification. I prefer just loop querying the price of each id one by one (possibly with caching of signal prices to avoid work repetition if there's performance concern). Let's talk about this tomorrow in person.